etc/try_parser.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+# Copyright 2023 The Servo Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import json
+from enum import Enum
+import sys
+
+
+class Layout(str, Enum):
+    none = 'none'
+    layout2013 = '2013'
+    layout2020 = '2020'
+    all = 'all'
+
+
+class OS(str, Enum):
+    linux = 'linux'
+    mac = 'mac'
+    win = 'windows'
+
+
+class JobConfig(object):
+    """
+    Represents one config tuple
+
+    name[os=_, layout=_, ...]
+    """
+
+    def __init__(self, name: str, os: OS = OS.linux, layout: Layout = Layout.none,
+                 profile: str = "release", unit_test: bool = True, wpt: str = ""):
+        self.os: OS = os
+        self.name: str = name
+        self.wpt_layout: Layout = layout
+        self.profile: str = profile
+        self.unit_tests: bool = unit_test
+        self.wpt_tests_to_run: str = wpt
+
+
+def preset(s: str) -> JobConfig | None:
+    s = s.lower()
+
+    if s == "linux":
+        return JobConfig("Linux", OS.linux)
+    elif s == "mac":
```

Comment:
I think it makes sense to accept "mac" as well as "macos" as you've done with "win" and "windows" below.